onnxruntime/core/providers/js/operators/batch_norm.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/providers/js/js_kernel.h"
+
+namespace onnxruntime {
+namespace js {
+
+template <typename T, bool is_channels_last>
+class BatchNorm final : public JsKernel {
+ public:
+  explicit BatchNorm(const OpKernelInfo& info) : JsKernel(info) {
+    float epsilon = info.GetAttrOrDefault<float>("epsilon", 1e-5);
+    float momentum = info.GetAttrOrDefault<float>("momentum", 0.9);
+    int64_t spatial = info.GetAttrOrDefault<int64_t>("spatial", 1);
+
+    const auto& node = info.node();
+    int opset = node.SinceVersion();
+    int64_t training_mode = opset <= 9 ? info.GetOutputCount() > 1 : info.GetAttrOrDefault<int64_t>("training_mode", 0);
```

Comment:
if we don't support training mode anyway, perhaps fail it here when training mode is ON and no need to pass to JS

Reply:
I add throw error [1] in js side. It's convenient to only change js side code if we want to support training mode in future.